object.c

```diff
@@ -268,21 +269,65 @@ rb_obj_singleton_class(VALUE obj)
 MJIT_FUNC_EXPORTED void
 rb_obj_copy_ivar(VALUE dest, VALUE obj)
```

Comment:
Good call, we had inadvertently changed the caller to condition on `if (!RB_TYPE_P(obj, T_CLASS) && !RB_TYPE_P(obj, T_MODULE))` instead of what it was. Changed it back, so we now haven't changed any callers of `rb_obj_copy_ivar`, it's still specific to `T_OBJECT`